### DISCUSSION TITLE
Feature Request: iOS Share Sheet Integration

### DISCUSSION THREAD

**Discussion by hnt3b453** [balzack/databag#21] [Ideas]
Another feature I'd love to see in the iOS app is the ability to use the share sheet to share links/photos/etc....

Keep up the great work!

**Comment by balzack** [comment]
[edit] I googled and now I know what a share sheet is.

Also great feedback. I'll need to research how to do this properly, but it seems like pretty important functionality.

**Reply by hnt3b453** [reply]
That's awesome! Thank you! I agree it's pretty important, and is probably the key feature I'd need to switch over entirely from Matrix.

I just want to say again, I love how lightweight and easy to implement Databag is. Matrix has been pretty good, but it's pretty dang complicated to get set up and maintain.

**Comment by balzack** [comment]
This feature ended up being pretty straight forward to implement. I'll do a bit more testing and probably push and iOS update over the weekend. It will be a first pass where the UI and formatting may need some additional tweaking.

I really enjoy working on the project, kinda my favorite thing to do, but it feels great when other people appreciate it as well. Thanks for the compliment!

**Reply by hnt3b453** [reply]
You're welcome! I'll keep an eye out for the update!

**Comment by balzack** [comment]
i just realized i'm probably only implementing 1/2 of the feature at the moment. I was focusing on sharing a message from databag on some other app, (for example: someone clicks a share button on a message, and then can send it over iMessage).

I imagine you are interested in the reverse, which might be a lot more tricky. Is this first 1/2 implementation of value?

**Reply by hnt3b453** [reply]
The second implementation of sharing items from other apps to databag (from the browser (I use Firefox Focus), Apollo/Reddit, Photos.app, and the like) would be far more useful. 

Sharing messages out of Databag would IMO be of fairly low value.

**Reply by balzack** [reply]
yea, that is what I figured. okay, i probably wont be able to release this feature over the weekend. I will probably have to implement the feature of 'clicking on a notification and bringing you to the conversation thread' first. I think this logic is a prerequisite of proper share sheet integration. thanks again for the valuable feedback.

**Reply by hnt3b453** [reply]
No problem! Thank you again for your amazing work. Is there anything else I can do to help/support you in the work you're doing?

**Reply by balzack** [reply]
Honestly what you are already doing for the project is so important. I plan to spend the next several months implementing the various feature requests. 

I don't want to exclude you if you have any specific interests. I know eventually the project will need a lot of work in other areas, but I don't see those in need of immediate attention. Do you see anything that should be done beyond adding features for now?

[edit] 
I have been told the UI is not intuitive at parts. If you or the people you communicate with have suggestions, that would also be helpful.

**Reply by hnt3b453** [reply]
Awesome. I'm happy to help with the project! I also appreciate the development of the features I've been asking for. I just don't want to come off as entitled or whatever when it comes to feature requests. :) That's definitely not the case, I understand in all cases it'll take time.

I'll definitely look at the UI and will ask the people I communicate with for UI feedback as well. I'm more tech savvy as it were, so for me I didn't find the UI unintuitive, but I can see how for less tech savvy people it could be, so I'll try to get any feedback I can from that perspective.

**Reply by balzack** [reply]
I have an off topic question. Currently e2e is not enabled by default. The user has to setup a key in the profile, and then switch 'sealed channel' to on when starting a topic. On those sealed topics, photos and videos are not yet supported. Is this a deal breaker for you or people you know?

**Comment by balzack** [comment]
@hnt3b453 I updated the iOS version to support this feature. I opted to avoid 'deep linking' until i implement that the feature of linking a topic to push notification. As a result, when you use the share sheet with databag, you will be presented with a list of topics with which you can share. Let me know if it's too clunky compared to apps like 'slack', which let you specify the specific channel from the share sheet.

[edit] I just noticed the 'sealed' topics do show up for sharing, but because they aren't supported yet on a sealed channel, it won't let you send. On that note, I think i finally have a good compromise for how to implement e2e with all the currently supported media types { image, audio, and video }. I should have it working in roughly 2-3 weeks. 

As always, thanks for your support!
